rules/alerting.go

```diff
@@ -404,6 +407,15 @@ func (r *AlertingRule) Eval(ctx context.Context, ts time.Time, query QueryFunc,
 		}
 	}
 
+	// Check that the alert does not produce identical metrics after applying
```

Comment:
What happens to h.active etc. above? We should probably do this check before all that, so this has the same effect as a query failure.

Reply:
@brian-brazil in that case I completely changed the way we check for duplicate alerts